nerfstudio/cameras/camera_optimizers.py

```diff
@@ -157,15 +157,15 @@ def apply_to_camera(self, camera: Cameras) -> torch.Tensor:
         if self.config.mode == "off":
             return camera.camera_to_worlds
 
-        assert camera.metadata is not None, "Must provide id of camera in its metadata"
-        if "cam_idx" not in camera.metadata:
-            # Evalutaion cams?
+        if camera.metadata is None or "cam_idx" not in camera.metadata:
+            # Viser cameras
             return camera.camera_to_worlds
 
         camera_idx = camera.metadata["cam_idx"]
-        adj = self(torch.tensor([camera_idx], dtype=torch.long, device=camera.device))  # type: ignore
-        adj = torch.cat([adj, torch.Tensor([0, 0, 0, 1])[None, None].to(adj)], dim=1)
-        return torch.bmm(camera.camera_to_worlds, adj)
+        adj = self(torch.tensor([camera_idx], dtype=torch.long)).to(camera.device)  # type: ignore
+        c2w4x4 = torch.cat([camera.camera_to_worlds, torch.Tensor([0, 0, 0, 1])[None, None].to(camera.device)], dim=1)
+        # apply correction in world coordinates so that the adjustments are consistent with apply_to_raybundle
+        return torch.bmm(adj, c2w4x4)
```

Comment:
Based on #3073 's discussion, the apply_to_raybundle is neither multiplying transform from left or right. I believe, one should return
```
rotations = torch.bmm(adj[..., :3, :3], camera.camera_to_worlds[..., :3, :3])
translations = camera.camera_to_worlds[..., :3, 3] + adj[..., :3, 3]
```

Reply:
Thanks for checking. You're right... Quite weird that this actually worked quite well in [my test](https://github.com/nerfstudio-project/nerfstudio/pull/3116#discussion_r1615140185). Perhaps the angular adjustments were so small in the synthetic test that the difference between the two formulas was negligible. I'll anyway change it to match...

Reply:
@jb-ye Now actually fixed. Here's my test results that show the subtle difference
 1. Artificial pose noise, Splatfacto reconstruction without pose optimization
 2. Poses optimized with Nerfacto, reconstruction with Splatfacto without pose opt. *old version*
 3. Same as above but with the new version


### 1

![no-pose-opt](https://github.com/nerfstudio-project/nerfstudio/assets/1812432/3f5811ae-177e-41af-b462-8386c2f5a23a)

### 2

![old](https://github.com/nerfstudio-project/nerfstudio/assets/1812432/db87df1d-2e75-412d-9e8d-fdba274e84af)

### 3

![new](https://github.com/nerfstudio-project/nerfstudio/assets/1812432/176ab184-6261-4601-8506-ba5f4359dd86)